GH-13376: Improve community server links and phrasing in README.md to be more inviting. (#13379)
mattermost/mattermost-server: README.md

## Try out Mattermost

> - [Join the Mattermost Contributor's server](https://community.mattermost.com/) (latest nightly builds, unstable. Get into contact with us and our community here.)

Reviewer: ```suggestion
- [Join the Mattermost Contributor's server](https://community.mattermost.com/signup_user_complete/?id=f1924a8db44ff3bb41c96424cdc20676) to join community discussions about contributions, development and more

Reviewer: The community.mattermost.com is now also on the latest stable version, so no need to call it out specifically

> - [Join the Mattermost Demo server](https://demo.mattermost.com/signup_email) (latest stable version. Use this server to generally play with Mattermost and take everything for a spin.)

Reviewer: ```suggestion
- [Join the Mattermost Demo server](https://demo.mattermost.com/signup_email) to try out Mattermost and explore sample use cases
```

- [Contribute Code](https://developers.mattermost.com/contribute/getting-started/)
- [Find "Help Wanted" projects](https://github.com/mattermost/mattermost-server/issues?page=1&q=is%3Aissue+is%3Aopen+%22Help+Wanted%22&utf8=%E2%9C%93)

> - [Join Developer Discussion on a Mattermost Server for contributors](https://community.mattermost.com/signup_user_complete/?id=f1924a8db44ff3bb41c96424cdc20676)

Author: The other occurrences link to the server blank, this one linkts to the `/signup_user_complete/?id=...` part. Is the id relevant to track sign-ups from the GitHub repo or just an artifact? 
In case it's meant for tracking it should probably be put into every occurence, in the other case I think it should probably be discarded. What do you think?

Reviewer: @lenucksi the ID ensures that they get automatically invited to the Contributors team when they sign up, and don't have to explicitly join a team. Saves new community members a step. Not used for tracking

But agree, we can use the same link for both occurrences

Author: Thanks for the insight!

- **Twitter** - Follow [Mattermost](https://twitter.com/mattermost)
- **Blog** - Get the latest updates from the [Mattermost blog](https://mattermost.com/blog/).
- **Email** - Subscribe to our [newsletter](http://mattermost.us11.list-manage.com/subscribe?u=6cdba22349ae374e188e7ab8e&id=2add1c8034) (1 or 2 per month)

> - **Mattermost** - Join the #contributors channel on [the Mattermost Community Server](https://community.mattermost.com/). 

Reviewer: ```suggestion
- **Mattermost** - Join the ~contributors channel on [the Mattermost Community Server](https://community.mattermost.com/signup_user_complete/?id=f1924a8db44ff3bb41c96424cdc20676). 
```

> - **IRC** - Join the #matterbridge channel on [Freenode](https://freenode.net/) (thanks to [matterircd](https://github.com/42wim/matterircd))

Reviewer 2: ```suggestion
- **IRC** - Join the #matterbridge channel on [Freenode](https://freenode.net/) (thanks to [matterircd](https://github.com/42wim/matterircd)).
```
